XenAdmin/Core/HealthCheck.cs

```diff
@@ -75,22 +71,49 @@ public static void CheckForAnalysisResults()
         }
 
         /// <summary>
-        /// Checks the analysis results for the specified connections
+        /// Checks the analysis results for the specified connections. 
+        /// Will only perform the check if the connection has sufficient rights; will not show the action progress.
         /// </summary>
         public static void CheckForAnalysisResults(object connection)
         {
-            Pool pool = Helpers.GetPoolOfOne((IXenConnection)connection);
-            if (pool == null || Helpers.FeatureForbidden((IXenConnection)connection, Host.RestrictHealthCheck))
-                return;
+            CheckForAnalysisResults(connection, true);
+        }
+        
+        /// <summary>
+        /// Checks the analysis results for the specified connections.
+        /// Will only perform the check if the connection has sufficient rights; If suppressHistory is false, it will show the action progress.
+        /// </summary>
+        public static void CheckForAnalysisResults(object connection, bool suppressHistory)
+        {
+            var action = GetAction((IXenConnection)connection, suppressHistory);
+            if (action != null)
+            {
+                action.Completed += actionCompleted;
+                action.RunAsync();
+            }
+        }
+
+        private static AsyncAction GetAction(IXenConnection connection, bool suppressHistory)
+        {
+            Pool pool = Helpers.GetPoolOfOne(connection);
+            if (pool == null || Helpers.FeatureForbidden(connection, Host.RestrictHealthCheck))
+                return null;
 
             var healthCheckSettings = pool.HealthCheckSettings;
 
             if (healthCheckSettings.Status == HealthCheckStatus.Enabled && !healthCheckSettings.HasAnalysisResult)
             {
-                var action = new GetHealthCheckAnalysisResultAction(pool, Registry.HealthCheckDiagnosticDomainName, true);
-                action.Completed += actionCompleted;
-                action.RunAsync();
+                var action = new GetHealthCheckAnalysisResultAction(pool, Registry.HealthCheckDiagnosticDomainName, suppressHistory);
+
+                if (PassedRbacChecks(pool.Connection))
+                    return action;
```

Comment:
I think we should create the action only if the connection passes the Rbac checks.